tests/run/rescue.scala

```diff
@@ -0,0 +1,18 @@
+object lib {
+  inline def (op: => T) rescue[T] (fallback: => T) = try op catch { case _: Throwable => fallback }
+  inline def (op: => T) rescue[T, E <: Throwable] (fallback: E => T) = try op catch { case ex: E => fallback(ex) }
```

Comment:
since E is erased to Throwable, this will pass all exceptions to fallback, which might explode on some of them, if you take a partial function you could do instead something like:
```suggestion
  inline def (op: => T) rescue[T, E <: Throwable] (fallback: PartialFunction[E, T]) =
    try op
    catch {
      case ex: E =>
        if (fallback.isDefinedAt(ex)) fallback(ex) else throw ex
    }
```

Reply:
Ah, I guess it won't be erased like this due to inlining, ignore me then :).

Reply:
If you flip the order of the tparams, could you call it as

```scala
riskyOp() recover[IOException] { ex => ... }
```

rather than ascribing the type of `ex`?

Reply:
The parser is not happy with the syntax.

In the latest commit, the suggestion by @smarter is adopted. The original API is problematic if users want to catch several kinds of errors. In particular, the following code will fail:

```Scala
 assert(
    {
      {
        val a = 9 / 0 rescue {
          case ex: NullPointerException => 4
          case ex: ClassCastException => 5
        }
        a * a
      } rescue {
        case ex: ArithmeticException => 3
      }
    } == 3
  )
```